### PR TITLE
PR for glibc

### PR DESCRIPTION
**Pull request by robinyuan1002**
Mainly add MAKE_SYSCALL to file in glibc/sysdeps.

### CODE REVIEW

File: io/fcntl.c [hunk 1]
```diff
@@ -27,9 +27,14 @@ __fcntl (int fd, int cmd, ...)
       __set_errno (EBADF);
       return -1;
     }
+  
+  // va_list args;
```
Comment (yzhang71) [inline io/fcntl.c]: Please remove the redundant codes

File: posix/execv.c [hunk 2]
```diff
@@ -16,11 +16,22 @@
    <https://www.gnu.org/licenses/>.  */
 
 #include <unistd.h>
+#include <syscall-template.h>
 
+// __int32_t __imported_wasi_execv() __attribute__((
+//     __import_module__("wasix"),
+//     __import_name__("lind-execv")
```
Comment (yzhang71) [inline posix/execv.c]: Please remove the redundant codes

File: lind_syscall/crt1/crt1.c
Comment (Yaxuan-w) [inline lind_syscall/crt1/crt1.c:1]: Is `crt1.c` file still the actual entry point of the program? The `_start` or `__start` function is the entry point and finally calls the user-defined `main()` function, but I don't see `__exit`. Where do we handle `__exit`?

File: lind_syscall/crt1_old.o
Comment (Yaxuan-w) [inline lind_syscall/crt1_old.o:1]: why we remain old version here?

File: lind_syscall/lind_syscall.c
Comment (Yaxuan-w) [inline lind_syscall/lind_syscall.c:1]: That would be great to add comments at very beginning of this file

File: posix/fork.c [hunk 3]
```diff
@@ -39,98 +52,107 @@ fresetlockfiles (void)
...
-    {
-      call_function_static_weak (__nss_database_fork_prepare_parent,
-				 &nss_database_data);
+  // /* If we are not running multiple threads, we do not have to
+  //    preserve lock state.  If fork runs from a signal handler, only
+  //    async-signal-safe functions can be used in the child.  These data
+  //    structures are only used by unsafe functions, so their state does
+  //    not matter if fork was called from a signal handler.  */
+  // if (multiple_threads)
+  //   {
```
Comment (Yaxuan-w) [inline posix/fork.c]: Do we need to remove those lines?

File: posix/getpid.c [hunk 4]
```diff
@@ -17,13 +17,25 @@
...
+#include <syscall-template.h>
+
+// int32_t __imported_wasi_getpid() __attribute__((
+//     __import_module__("wasix"),
+//     __import_name__("lind-getpid")
+// ));
+
+// int32_t __wasi_getpid() {
+//     return __imported_wasi_getpid(0);
+// }
```
Comment (Yaxuan-w) [inline posix/getpid.c]: Should we remove?

File: posix/getpid.c [hunk 5]
```diff
@@ -17,13 +17,25 @@
...
 
 /* Get the process ID of the calling process.  */
 int
 __getpid (void)
 {
-  __set_errno (ENOSYS);
-  return -1;
+  // __set_errno (ENOSYS);
+  // return -1;
+  // return __wasi_getpid();
```
Comment (Yaxuan-w) [inline posix/getpid.c]: same

File: sysdeps/posix/flock.c [hunk 6]
```diff
@@ -23,34 +23,36 @@
...
+  //     break;
+  //   default:
+  //     __set_errno (EINVAL);
+  //     return -1;
+  //   }
+
+  // lbuf.l_whence = SEEK_SET;
+  // lbuf.l_start = lbuf.l_len = 0L; /* Lock the whole file.  */
+
+  // return __fcntl (fd, (operation & LOCK_NB) ? F_SETLK : F_SETLKW, &lbuf);
```
Comment (Yaxuan-w) [inline sysdeps/posix/flock.c]: same

File: sysdeps/unix/sysv/linux/ifaddrs.c [hunk 7]
```diff
@@ -829,6 +830,7 @@ __getifaddrs (struct ifaddrs **ifap)
 
   do
     res = getifaddrs_internal (ifap);
+	// res = MAKE_SYSCALL(146, "syscall|getifaddrs", (uint64_t) ifap, NOTUSED, NOTUSED, NOTUSED, NOTUSED, NOTUSED);
```
Comment (Yaxuan-w) [inline sysdeps/unix/sysv/linux/ifaddrs.c]: should we remove here as well?

File: tests/access.wasm
Comment (Yaxuan-w) [inline tests/access.wasm:1]: Why do we need include wasm binaries? Could we generate it while testing?

File: tests/lind.metadata
Comment (Yaxuan-w) [inline tests/lind.metadata:1]: I'm a bit confused... Should we generate Lind's metadata in usage instead of pre-build? Or is there any specific reason that we have to do this way?

File: tests/mmap_munmap.wasm
Comment (Yaxuan-w) [inline tests/mmap_munmap.wasm:1]: Same with `access.wasm`

File: tests/open.wasm
Comment (Yaxuan-w) [inline tests/open.wasm:1]: Same with `access.wasm`

File: tests/pread.wasm
Comment (Yaxuan-w) [inline tests/pread.wasm:1]: Same with `access.wasm`

File: tests/read.wasm
Comment (Yaxuan-w) [inline tests/read.wasm:1]: Same with `access.wasm`

**Review by Yaxuan-w — CHANGES_REQUESTED** [review]
Awesome work! Minor suggestions

File: csu/wasm32/set_stack_pointer.s [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+	.type	set_stack_pointer,@function
+
+set_stack_pointer:
+	.functype	set_stack_pointer (i32) -> ()
+
+	local.get   0  # start_arg
+	i32.load    0  # stack
+	global.set  __stack_pointer
+
+	end_function
```
Comment (rennergade) [inline csu/wasm32/set_stack_pointer.s]: add newlines

File: nptl/pthread_create.c [hunk 9]
```diff
@@ -495,27 +489,27 @@ start_thread (void *arg)
...
+      }
       else
-	ret = pd->start_routine (pd->arg);
+        ret = pd->start_routine (pd->arg);
       THREAD_SETMEM (pd, result, ret);
     }
 
   /* Call destructors for the thread_local TLS variables.  */
-  call_function_static_weak (__call_tls_dtors);
+  // call_function_static_weak (__call_tls_dtors);
```
Comment (rennergade) [inline nptl/pthread_create.c]: anytime were commenting out original libc code we need a note for why

File: posix/fork.c (hunk 3 above)
Comment (rennergade) [reply posix/fork.c]: I think for the libc code we don' necessarily want to remove the commented original libc code but we want to leave notes for why we are disabling it.

File: stdlib/exit.c [hunk 10]
```diff
@@ -128,7 +137,7 @@ __run_exit_handlers (int status, struct exit_function_list **listp,
   if (run_list_atexit)
     call_function_static_weak (_IO_cleanup);
 
-  _exit (status);
+	__wasi_exit(status);
```
Comment (rennergade) [inline stdlib/exit.c]: comment, what is this

File: sysdeps/unix/sysv/linux/i386/clone.c [hunk 11]
```diff
@@ -1,4 +1,53 @@
-int __GI___clone (void) {
-  return 0;
+#include <sched.h>
+#include "libioP.h"
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/i386/clone.c]: this file needs a bunch of comments

File: target/bin/ldd [hunk 12]
```diff
@@ -24,9 +24,9 @@
 
 # We should be able to find the translation right at the beginning.
 TEXTDOMAIN=libc
-TEXTDOMAINDIR=/home/dennis/Documents/lind-wasm/glibc/target/share/locale
```
Comment (rennergade) [inline target/bin/ldd]: we probably don't want absolute paths here pointing to Dennis's home folder?

Comment (rennergade) [reply target/bin/ldd]: Same for a lot of these. Would be good to add comments for modified build files.

File: tests/access.wasm
Comment (robinyuan1002) [reply tests/access.wasm:1]: Yes we can, I will delete all of them

Comment (robinyuan1002) [reply tests/access.wasm:1]: I will remove this tests folder. Since I used Lind project's tests to test them.

File: sysdeps/unix/sysv/linux/i386/clone.c (hunk 11 above)
Comment (robinyuan1002) [reply sysdeps/unix/sysv/linux/i386/clone.c]: This is no longer useful anymore, I will restore to the original one

File: lind_syscall/crt1/crt1.c
Comment (qianxichen233) [reply lind_syscall/crt1/crt1.c:1]: We currently do not have anything to cleanup once original_main returns. So we just let _start function returns and the wasm module would exit normally

File: target/bin/ldd (hunk 12 above)
Comment (robinyuan1002) [reply target/bin/ldd]: This line is deleted, and I will comment for absolute paths

File: lind_syscall/crt1/crt1.c [hunk 13]
```diff
@@ -1,39 +1,117 @@
-#include <wasi/api.h>
 #include <stdlib.h>
 #include <sysexits.h>
 
-int32_t __imported_wasi_snapshot_preview1_args_sizes_get(int32_t arg0, int32_t arg1) __attribute__((
+// char** __wasilibc_environ;
+// weak_alias(__wasilibc_environ, _environ);
+// weak_alias(__wasilibc_environ, environ);
```
Comment (JustinCappos) [inline lind_syscall/crt1/crt1.c]: Let's not have commented code in the codebase unless there is a good reason

File: lind_syscall/crt1/crt1.c [hunk 14]
```diff
@@ -1,39 +1,117 @@
...
-__wasi_errno_t __wasi_args_sizes_get(
-    __wasi_size_t *retptr0,
-    __wasi_size_t *retptr1
+unsigned short __wasi_args_sizes_get(
+    unsigned long *retptr0,
+    unsigned long *retptr1
 ){
-    int32_t ret = __imported_wasi_snapshot_preview1_args_sizes_get((int32_t) retptr0, (int32_t) retptr1);
-    return (uint16_t) ret;
+    signed int ret = __imported_wasi_snapshot_preview1_args_sizes_get((signed int) retptr0, (signed int) retptr1);
```
Comment (JustinCappos) [inline lind_syscall/crt1/crt1.c]: Why are we changing the type definitions?  If the symbol isn't defined, isn't it easier to define it?

File: locale/localeconv.c [hunk 15]
```diff
@@ -25,42 +25,45 @@ __localeconv (void)
...
-  result.positive_sign = (char *) _NL_CURRENT (LC_MONETARY, POSITIVE_SIGN);
-  result.negative_sign = (char *) _NL_CURRENT (LC_MONETARY, NEGATIVE_SIGN);
+//   result.int_curr_symbol = (char *) _NL_CURRENT (LC_MONETARY, INT_CURR_SYMBOL);
+//   result.currency_symbol = (char *) _NL_CURRENT (LC_MONETARY, CURRENCY_SYMBOL);
+//   result.mon_decimal_point = (char *) _NL_CURRENT (LC_MONETARY,
+// 						   MON_DECIMAL_POINT);
+//   result.mon_thousands_sep = (char *) _NL_CURRENT (LC_MONETARY,
+// 						   MON_THOUSANDS_SEP);
+//   result.mon_grouping = (char *) _NL_CURRENT (LC_MONETARY, MON_GROUPING);
+//   if (*result.mon_grouping == '\177' || *result.mon_grouping == '\377')
```
Comment (JustinCappos) [inline locale/localeconv.c]: similar here.  Why comment this out?

File: nptl/pthread_cond_signal.c [hunk 16]
```diff
@@ -26,6 +26,7 @@
 
 #include <shlib-compat.h>
 #include <stap-probe.h>
+#include <stdio.h>
```
Comment (JustinCappos) [inline nptl/pthread_cond_signal.c]: Why is this needed all of the sudden?

File: nptl/pthread_create.c [hunk 17]
```diff
@@ -645,9 +651,9 @@ start_thread (void *arg)
 
      The exit code is zero since in case all threads exit by calling
      'pthread_exit' the exit status must be 0 (zero).  */
-
+  
+  pd->tid = 0;
```
Comment (JustinCappos) [inline nptl/pthread_create.c]: why is this reordered?

File: nptl/pthread_create.c [hunk 18]
```diff
@@ -342,14 +345,17 @@ static int create_thread (struct pthread *pd, const struct pthread_attr *attr,
...
   memset(args, 0, sizeof(struct clone_args));
   args->flags = clone_flags;
   args->stack = stackaddr;
-  args->stack_size = 65664;
+  args->stack = stackaddr + pd->stackblock_size - sizeof(struct clone_args) - TLS_TCB_SIZE;
+  args->stack_size = 16416 - sizeof(struct clone_args) - TLS_TCB_SIZE;
   args->child_tid = &pd->tid;
 
   int ret = __clone_internal(args, &start_thread, pd);
+  printf("pthread_create, child tid=%d, pd=%d\n", pd->tid, pd);
```
Comment (JustinCappos) [inline nptl/pthread_create.c]: Is this supposed to be added?

**Review by JustinCappos — CHANGES_REQUESTED** [review]
This PR is huge!   Can you break it up?

Also, a lot of commented code is included here.   Why is this?

Can you comment your code to explain why you're doing what you are doing?

File: locale/localeconv.c (hunk 15 above)
Comment (qianxichen233) [reply locale/localeconv.c]: Also added as comment: locale related stuff is not working currently since a lot of locale symbol is not found by wasm-ld. I currently havn't looked into the root reason for this, but this feature does not seem to be important right now so we might look into this later if we want to support it in the future

**Comment by qianxichen233** [comment]
The PR is splitted into three parts. This PR is specifically for adding syscalls. Other PRs are [adding includes](https://github.com/Lind-Project/glibc/pull/7) and [adding multi-process support](https://github.com/Lind-Project/glibc/pull/6). Could you review this PR again @JustinCappos , @Yaxuan-w, @rennergade, @yzhang71

File: csu/inotify.c [hunk 19]
```diff
@@ -0,0 +1,13 @@
+#include <stdint.h>
+
+// Qianxi Edit: add these two function because coreutils needs them but
```
Comment (rennergade) [inline csu/inotify.c]: I'd tag this with BUG:

ie:

BUG: add these two function because coreutils needs them but... - Qianxi Chen

File: libio/genops.c [hunk 20]
```diff
@@ -344,7 +344,8 @@ _IO_doallocbuf (FILE *fp)
   if (fp->_IO_buf_base)
     return;
   if (!(fp->_flags & _IO_UNBUFFERED) || fp->_mode > 0)
-    if (_IO_DOALLOCATE (fp) != EOF)
+    // if (_IO_DOALLOCATE (fp) != EOF)
```
Comment (rennergade) [inline libio/genops.c]: remove commented code

File: libio/ioputs.c [hunk 21]
```diff
@@ -35,11 +35,7 @@ _IO_puts (const char *str)
   size_t len = strlen (str);
   _IO_acquire_lock (stdout);
 
-  if ((_IO_vtable_offset (stdout) != 0
-       || _IO_fwide (stdout, -1) == -1)
-      && _IO_sputn (stdout, str, len) == len
-      && _IO_putc_unlocked ('\n', stdout) != EOF)
-    result = MIN (INT_MAX, len + 1);
+  write(STDIN_FILENO, str, len);
```
Comment (rennergade) [inline libio/ioputs.c]: I believe we changed this because of lack of a vtable. can we comment this change? @yzhang71 I think you were working on this

File: lind_syscall/lind_syscall.c [hunk 22]
```diff
@@ -15,14 +16,25 @@
    You should have received a copy of the GNU Lesser General Public
    License along with the GNU C Library.  If not, see
    <https://www.gnu.org/licenses/>.  */
-int __imported_wasi_snapshot_preview1_lind_syscall(unsigned int callnumber, unsigned long long callname, unsigned long long arg1, unsigned long long arg2, unsigned long long arg3, unsigned long long arg4, unsigned long long arg5, unsigned long long arg6, int ret) __attribute__((
-    __import_module__("wasi_snapshot_preview1"),
-    __import_name__("lind_syscall")
+
+//Entry point for wasmtime
```
Comment (rennergade) [inline lind_syscall/lind_syscall.c]: need a more extensive comment here

File: lind_syscall/lind_syscall.c [hunk 23]
```diff
@@ -15,14 +16,25 @@
...
+int __imported_wasi_snapshot_preview1_lind_syscall(unsigned int callnumber, unsigned long long callname, unsigned long long arg1, unsigned long long arg2, unsigned long long arg3, unsigned long long arg4, unsigned long long arg5, unsigned long long arg6) __attribute__((
+    __import_module__("lind"),
+    __import_name__("lind-syscall")
 ));
 
+
+//Part of Marco MAKE_SYSCALL, take in the number of the syscall and the name of the syscall and 6 argument.
```
Comment (rennergade) [inline lind_syscall/lind_syscall.c]: *Macro

also needs a more verbose comment

File: stdlib/exit.c [hunk 24]
```diff
@@ -137,4 +165,4 @@ exit (int status)
 {
   __run_exit_handlers (status, &__exit_funcs, true, true);
 }
-libc_hidden_def (exit)
+libc_hidden_def (exit)
```
Comment (rennergade) [inline stdlib/exit.c]: theres a lot of files where we remove the newlines at the end, pedantic but should clean that up

**Review by rennergade — COMMENTED** [review]
In general this is great, some minor fixes. 

Agree with Justin that we don't want to commit the commented code, but theres a lot of situations where we may want that code later.

I think the best solution is to comment all of those places with:

```
Lind-Wasm: Original glibc code removed for compatibility
   to find original source code refer to (OUR GLIBC VERSION NUM) at (FILEPATH):(LINENO)
```

if theres a glibc repo up thats easy to have a link too that might be even better

File: sysdeps/unix/sysv/linux/stat64.c [hunk 25]
```diff
@@ -22,11 +22,14 @@
 #include <fcntl.h>
 #include <kernel_stat.h>
 #include <stat_t64_cp.h>
+#include <syscall-template.h>
 
 int
 __stat64_time64 (const char *file, struct __stat64_t64 *buf)
 {
-  return __fstatat64_time64 (AT_FDCWD, file, buf, 0);
+  // Qianxi Edit: we do not have fstatat syscall in rawposix
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/stat64.c]: BUG: .... - Qianxi Chen

File: sysdeps/unix/sysv/linux/open_nocancel.c [hunk 26]
```diff
@@ -39,7 +40,8 @@ __open_nocancel (const char *file, int oflag, ...)
       va_end (arg);
     }
 
-  return INLINE_SYSCALL_CALL (openat, AT_FDCWD, file, oflag, mode);
+  // Qianxi Edit:
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/open_nocancel.c]: Is there supposed to be more here? if so "BUG: ...  - Qianxi Chen"

File: sysdeps/unix/sysv/linux/open64.c [hunk 27]
```diff
@@ -38,8 +39,8 @@ __libc_open64 (const char *file, int oflag, ...)
       va_end (arg);
     }
 
-  return SYSCALL_CANCEL (openat, AT_FDCWD, file, oflag | O_LARGEFILE,
-			 mode);
+  // Qianxi Edit: pass down to rawposix
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/open64.c]: BUG: ... - Qianxi Chen

File: sysdeps/unix/sysv/linux/nanosleep.c [hunk 28]
```diff
@@ -36,7 +36,9 @@ libc_hidden_def (__nanosleep64)
 int
 __nanosleep (const struct timespec *req, struct timespec *rem)
 {
-  struct __timespec64 treq64, trem64;
+  // Qianxi Edit: have to initilize the value to 0, not sure why it
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/nanosleep.c]: BUG: ... - Qianxi Chen

File: sysdeps/unix/sysv/linux/lstat64.c [hunk 29]
```diff
@@ -22,11 +22,14 @@
 #include <fcntl.h>
 #include <kernel_stat.h>
 #include <stat_t64_cp.h>
+#include <syscall-template.h>
 
 int
 __lstat64_time64 (const char *file, struct __stat64_t64 *buf)
 {
-  return __fstatat64_time64 (AT_FDCWD, file, buf, AT_SYMLINK_NOFOLLOW);
+  // Qianxi Edit: we do not have fstatat syscall in rawposix
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/lstat64.c]: BUG: ... - Qianxi Chen

File: sysdeps/unix/sysv/linux/lstat.c [hunk 30]
```diff
@@ -19,12 +19,15 @@
...
 #include <fcntl.h>
 #include <kernel_stat.h>
+#include <syscall-template.h>
 
 #if !XSTAT_IS_XSTAT64
 int
 __lstat (const char *file, struct stat *buf)
 {
-  return __fstatat (AT_FDCWD, file, buf, AT_SYMLINK_NOFOLLOW);
+  // Qianxi Edit: we do not have fstatat syscall in rawposix
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/lstat.c]: BUG: ... - Qianxi Chen

File: sysdeps/unix/sysv/linux/getdents64.c [hunk 31]
```diff
@@ -29,7 +30,9 @@ __getdents64 (int fd, void *buf, size_t nbytes)
      checks in the kernel use an int type.  */
   if (nbytes > INT_MAX)
     nbytes = INT_MAX;
-  return INLINE_SYSCALL_CALL (getdents64, fd, buf, nbytes);
+  // Qianxi Edit: pass down to rawposix instead
+	return MAKE_SYSCALL(23, "syscall|getdents", (uint64_t) fd, (uint64_t) buf, (uint64_t) nbytes, NOTUSED, NOTUSED, NOTUSED);
+  // return INLINE_SYSCALL_CALL (getdents64, fd, buf, nbytes);
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/getdents64.c]: why commented?

File: sysdeps/unix/sysv/linux/getdents64.c [hunk 32]
```diff
@@ -29,7 +30,9 @@ __getdents64 (int fd, void *buf, size_t nbytes)
      checks in the kernel use an int type.  */
   if (nbytes > INT_MAX)
     nbytes = INT_MAX;
-  return INLINE_SYSCALL_CALL (getdents64, fd, buf, nbytes);
+  // Qianxi Edit: pass down to rawposix instead
```
Comment (rennergade) [inline sysdeps/unix/sysv/linux/getdents64.c]: Not sure what this one is

**Review by rennergade — CHANGES_REQUESTED** [review]
I tried to leave a review of places where it says "Qianxi Edit" also seeing some "Dennis Edit" comments.

Realistically, where its a bug we need to still fix or will look at later on it should be 

"BUG: DESCRIPTION - Author Name"

if its just marking a change it should just note that, theres a lot of these where its just a comment over the MAKE_SYSCALL macro. Those should just be like:

"Added MAKE_SYSCALL macro to interface with Lind - Author Name"

**Comment by qianxichen233** [comment]
> I tried to leave a review of places where it says "Qianxi Edit" also seeing some "Dennis Edit" comments.
> 
> Realistically, where its a bug we need to still fix or will look at later on it should be
> 
> "BUG: DESCRIPTION - Author Name"
> 
> if its just marking a change it should just note that, theres a lot of these where its just a comment over the MAKE_SYSCALL macro. Those should just be like:
> 
> "Added MAKE_SYSCALL macro to interface with Lind - Author Name"

Updated comments based on this. Could you review again

File: libio/ioputs.c (hunk 21 above)
Comment (rennergade) [reply libio/ioputs.c]: this still needs a comment. Can you check with @yzhang71

File: stdlib/atexit.c [hunk 33]
```diff
@@ -34,7 +34,39 @@
...
+#include <pointer_guard.h>
+
+int
+attribute_hidden
+__internal_atexit_2 (void (*func) (void),
+		   struct exit_function_list **listp)
+{
+  struct exit_function *new;
+
+  /* As a QoI issue we detect NULL early with an assertion instead
```
Comment (rennergade) [inline stdlib/atexit.c:51]: What is QoI?

**Review by rennergade — CHANGES_REQUESTED** [review]
OK I think this looks great. We need a comment for the vtable change Dennis added and I had a question about what one comment meant. Will merge after those are handled.

File: stdlib/atexit.c (hunk 33 above)
Comment (qianxichen233) [reply stdlib/atexit.c:51]: I am also not sure. This function is copied from __internal_atexit, and I only changed its logic at the end: `new->func.at = func; new->flavor = ef_at;`. The comment was from original __internal_atexit

File: libio/ioputs.c (hunk 21 above)
Comment (yzhang71) [reply libio/ioputs.c]: Yeah, this is a different issue with the previous vtable issue. The fix for this is here: https://github.com/Lind-Project/glibc/commit/1afde27bf804055e835ce76456e8fb5433263312

Comment (qianxichen233) [reply libio/ioputs.c]: fixed. This function is for `puts`. Previously I have to replace vtable with write_syscall. But since vtable is already fixed as dennis mentioned a couple of days ago, so this file could be changed back

File: nptl/futex-internal.c [hunk 34]
```diff
@@ -52,7 +52,7 @@ __futex_abstimed_wait_common32 (unsigned int* futex_word,
 }
 #endif /* ! __ASSUME_TIME64_SYSCALLS */
 
-// Dennis Edit: disable syscal cancel
+// BUG: disable syscal cancel - Dennis
```
Comment (rennergade) [inline nptl/futex-internal.c]: these both should say "syscall"

File: time/bits/types/struct_timespec.h [hunk 35]
```diff
@@ -8,26 +8,17 @@
...
-#if __WORDSIZE == 64 \
-  || (defined __SYSCALL_WORDSIZE && __SYSCALL_WORDSIZE == 64) \
-  || (__TIMESIZE == 32 && !defined __USE_TIME_BITS64)
-  __syscall_slong_t tv_nsec;	/* Nanoseconds.  */
-#else
-# if __BYTE_ORDER == __BIG_ENDIAN
-  int: 32;           /* Padding.  */
+  time_t tv_sec;		/* Seconds.  */
+  int: 32;
+  // __int32_t __padding;           /* Padding.  */
```
Comment (rennergade) [inline time/bits/types/struct_timespec.h]: why are these commented out?

**Review by rennergade — CHANGES_REQUESTED** [review]
sorry a few more little things to fix

File: time/bits/types/struct_timespec.h (hunk 35 above)
Comment (qianxichen233) [reply time/bits/types/struct_timespec.h]: I dont see any specific reason for this, I think int: 32 is equivelent with declare a __int32_t variable. I just reverted this part

**Review by rennergade — APPROVED** [review]
approved

File: lind_syscall/lind_syscall.o
Comment (rennergade) [inline lind_syscall/lind_syscall.o:1]: should this binary be here?

Comment (qianxichen233) [reply lind_syscall/lind_syscall.o:1]: This needs to be here currently for convenience. Since the building of lind_syscall.c hasn't been integrated into makefile yet, this file must be manually built right now. And also since lind_syscall.c shouldn't be changed frequently, so this binary is temporily left here.

Comment (rennergade) [reply lind_syscall/lind_syscall.o:1]: Ok, I guess fine for now. Can you open up an issue in the lind-wasm repo for someone to eventually fix this?
